### ISSUE TITLE
Suggestion: Remove responsibility for hierarchy validation from the provider

### ISSUE THREAD

**Issue by jaredfholgate** [Azure/terraform-provider-alz#24]
The provider currently has sophisticated capabilities to keep track of and validate if there are duplicate management groups names or multiple root management groups. This validation shows at apply time.

My suggestion is that although this capability is great, it should not be the responsibility of the provider `alz_archetype` data source to understand management groups outside of it's scope.

Reasons:

- There is a required `parent_id` field, that makes the graph generated by terraform have some dependencies that slows down the overall landing zone deployment. At the moment, this means that the apply happens layer by layer of the hierarchy and (subject to testing) I believe that more policy assignments could happen in parallel if this `parent_id` was removed.
- The individual `alz_archetype`, should ideally not have hidden dependencies on other instances of the `alz_archetype` data source. I believe this could cause confusion in the future as a data source is not expected to be dependent on state outside of it's input values.
- The check on a single root management group could potentially limit the flexibility of the module / provider. It may be the case that two root management groups is a valid design for a customer?
- Probably most importantly, removing these checks could simply the code base and make it easier to maintain and contribute to in the future.

In summary, can we remove this `parent_id` field and the associated logic behind it:

![image](https://github.com/Azure/terraform-provider-alz/assets/1612200/13ece3c6-47d9-4592-a4d5-6fb99fd9df20)

Very happy to discuss this with the team, as am very conscious that a lot of work has gone into this and my suggestion is to remove functionality. I am also very new to this, so very likely I am missing something obvious here.

**Comment by matt-FFFFFF** [comment]
One thing...

How do we ensure that the definition exists for a policy assignment at a child scope. Consider the case where the policy def is custom and needs to be created at a parent scope.

Could we handle this in the module, rather than the provider?

**Comment by jaredfholgate** [comment]
Good point and one I hadn't considered. Will have a think and can discuss next week.

**Comment by matt-FFFFFF** [comment]
Closing as architectures now in library
